src/controllers/QueryController.ts

```diff
@@ -174,6 +174,20 @@ export class QueryController extends RootComponent {
     return this.lastQueryResults;
   }
 
+  /**
+   * Returns the plan of execution of a search
+   * @returns {ExecutionPlan}
+   */
+  public async fetchQueryExecutionPlan() {
+    const query = this.createQueryBuilder(new DefaultQueryOptions()).build();
+    this.logger.debug('Fetching query execution plan');
+    try {
+      return await this.getEndpoint().plan(query);
```

Comment:
I think the `await` here might be redundant. If you remove it, you could also remove the `async` from this function.

Reply:
I believe it is needed. What if you call that method directly? You have to wait for /plan to be fetched.